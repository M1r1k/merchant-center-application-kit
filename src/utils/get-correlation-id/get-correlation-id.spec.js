import selectProjectKey from '../select-project-key';
import selectUserId from '../select-user-id';
import getCorrelationId from './get-correlation-id';

jest.mock('uuid/v4', () => () => 'test-uuid');
jest.mock('../select-project-key', () => jest.fn(() => 'test-project-key'));
jest.mock('../select-user-id', () => jest.fn(() => 'test-user-id'));

describe('getCorrelationId', () => {
  let correlationId;

  beforeEach(() => {
    correlationId = getCorrelationId();
  });

  it('should match snapshot', () => {
    expect(correlationId).toMatchSnapshot();
  });

  it('should invoke `selectUserId`', () => {
    expect(selectUserId).toHaveBeenCalled();
  });

  it('should invoke `selectProjectKey`', () => {
    expect(selectProjectKey).toHaveBeenCalled();
  });

  describe('without `userId`', () => {
    beforeEach(() => {
      selectUserId.mockReturnValue(null);
      correlationId = getCorrelationId();
    });

    it('should match snapshot', () => {
      expect(correlationId).toMatchSnapshot();
    });

    it('should not contain `null`', () => {
      // NOTE: `'null'` would be stringified.
      expect(correlationId).not.toEqual(expect.stringContaining('null'));
    });
  });
});
