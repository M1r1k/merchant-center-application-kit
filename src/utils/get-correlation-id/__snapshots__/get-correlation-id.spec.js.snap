// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCorrelationId should match snapshot 1`] = `"mc/test-project-key/test-user-id/test-uuid"`;

exports[`getCorrelationId without \`userId\` should match snapshot 1`] = `"mc/test-project-key/test-uuid"`;
